galeracluster/source/faq.rst

```diff
@@ -5,24 +5,113 @@
+
+
+
+.. rubric:: How Do I Upgrade the Cluster?
+.. _`faq-upgrade`:
+
+Periodically, updates will become available for Galera Cluster.  The database server itself or the :term:`Galera Replication Plugin`.  To update the software for the node, complete the following steps:
+
+#. Stop the node.
+#. Upgrade the software.
+#. Restart the node.
+
+In addition to this, you also need to transfer client connections from node you want to upgrade to another node for the duration of the migration.
+
+.. seealso:: For more information on upgrade process, see :doc:`upgrading`.
+
+.. rubric:: What InnoDB Isolation Levels does Galera Cluster Support?
+.. _`faq-isolation-levels`:
+
+You can use all isolation levels.  Locally, in a given node, transaction isolation works as it does natively with InnoDB.
+
+That said, globally, with transactions processing in separate node Galera Cluster implements transaction level called ``SNAPSHOT ISOLATION``.  The ``SNAPSHOT ISOLATION`` level occurs between ``REPEATABLE READ`` and ``SERIALIZABLE`` levels.
+
+The ``SERIALIZABLE`` level cannot be guaranteed in the multi-master use case, because Galera Cluster replication does not carry a transaction read set.  Also, ``SERIALIZABLE`` transaction is vulnerable to multi-master conflicts.  It holds read locks and any replicated write to read locked row will cause the transaction to abort.  Hence, it is recommended not to use it in Galera Cluster.
+
+.. seealso:: For more information, see :doc:`isolationlevels`.
+
+	     
+.. rubric:: How are DDL's Handled by Galera Cluster?
+.. _`ddl-galera`:
+
+For :abbr:`DDL (Data Definition Language)` statements and similar queries, Galera Cluster has two modes of execution:
+
+- :term:`Total Order Isolation` Where the query is replicated in a statement before executing on the master.  The node waits for all preceding transactions to commit and then all nodes simultaneously execute the transaction in isolation.
+
+- :term:`Rolling Schema Upgrade` Where the schema upgrades run locally, blocking only the node on which they are run.  The changes do not replicate to the rest of the cluster.
+
+.. seealso:: For more information, see :doc:`schemaupgrades`.
   
-      110906 17:45:01 [Note] WSREP: BF kill (1, seqno: 16962377), victim:  (140588996478720 4) trx: 35525064
-      110906 17:45:01 [Note] WSREP: Aborting query: commit
-      110906 17:45:01 [Note] WSREP: kill trx QUERY_COMMITTING for 35525064
-      110906 17:45:01 [Note] WSREP: commit failed for reason: 3, seqno: -1
+	     
+
+
+.. rubric:: What if connections give an ``Unknown command`` error?
+.. _`connection-unknown-command`:
+
+Your cluster experiences a temporary split, during which a portion of the nodes lose connectivity to the :term:`Primary Component`.  When it reconnects, nodes from the former nonoperational component drop database client connections.  New connections to the database client returns ``Unknown command`` errors.

```

Comment:
"lose" -> loses
"it reconnects" -> they reconnect
"drop database client connections" -> drop their client connections
"returns" -> return